server/routes/api/campaigns.js

```diff
@@ -252,4 +256,160 @@ router.delete(
   }
 );
 
+// @route POST /api/campaigns/:campaignId/steps
+// @desc Add a step to a campaign. Requires step name in request body.
+// @access Authenticated Users
+router.post(
+  "/:campaignId/steps",
+  passport.authenticate("jwt", { session: false }),
+  async (req, res) => {
+    const { errors, isValid } = validateStepInput(req.body);
+    if (!isValid) {
+      return res.status(400).json(errors);
+    }
+
+    const campaignId = req.params.campaignId;
+    const { name } = req.body;
+
+    try {
+      const campaign = await Campaign.findById(campaignId);
```

Comment:
Recommend we should check the campaign is owned by logged in user (req.user.id) so that logged in users can't add steps to other people's campaigns. See above edit/delete routes for examples.